crates/uv-toolchain/src/discovery.rs

```diff
@@ -1495,75 +1474,49 @@ impl fmt::Display for ToolchainSource {
 
 impl fmt::Display for ToolchainPreference {
     fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
-        match self {
-            Self::OnlyManaged => f.write_str("managed toolchains"),
-            Self::OnlySystem => f.write_str("system toolchains"),
-            Self::PreferInstalledManaged | Self::PreferManaged | Self::PreferSystem => {
-                f.write_str("managed or system toolchains")
+        let s = match self {
+            Self::OnlyManaged => "managed toolchains",
+            Self::PreferManaged | Self::PreferInstalledManaged | Self::PreferSystem => {
+                if cfg!(windows) {
+                    "managed toolchains, system path, or `py` launcher"
+                } else {
+                    "managed toolchains or system path"
+                }
             }
-        }
+            Self::OnlySystem => {
+                if cfg!(windows) {
+                    "system path or `py` launcher"
+                } else {
+                    "system path"
+                }
+            }
+        };
+        f.write_str(s)
     }
 }
 
 impl fmt::Display for ToolchainNotFound {
-    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
-        match self {
-            Self::NoPythonInstallation(sources, None | Some(VersionRequest::Any)) => {
-                write!(f, "No Python interpreters found in {sources}")
-            }
-            Self::NoPythonInstallation(sources, Some(version)) => {
-                write!(f, "No Python {version} interpreters found in {sources}")
-            }
-            Self::NoMatchingVersion(sources, VersionRequest::Any) => {
-                write!(f, "No Python interpreter found in {sources}")
-            }
-            Self::NoMatchingVersion(sources, version) => {
-                write!(f, "No interpreter found for Python {version} in {sources}")
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let sources = match self.environment_preference {
+            EnvironmentPreference::Any => {
+                format!("virtual environments or {}", self.toolchain_preference)
```

Comment:
Would this be "virtual environments or managed or system toolchains"? I feel like it should be Oxford comma'd but I know that's tedious...

Reply:
😬 I know right. I considered doing an extra match to fix this.... but I didn't have it in me. Think it's worth it? I might want to construct a test covering this so I know it matters if I do it.